datamap/mv/core/src/main/scala/org/apache/carbondata/mv/rewrite/DefaultMatchMaker.scala

```diff
@@ -354,6 +354,30 @@ object SelectSelectNoChildDelta extends DefaultMatchPattern with PredicateHelper
 }
 
 object GroupbyGroupbyNoChildDelta extends DefaultMatchPattern {
+  def  tryMatch(gb_2a: GroupBy,
+               gb_2q: GroupBy,
+               isGroupingEmR: Boolean,
+                isInheritTableRelation: Boolean) : Seq[ModularPlan] = {
+    val aliasMap = AttributeMap(gb_2a.outputList.collect { case a: Alias =>
+      (a.toAttribute, a)})
+    if (isGroupingEmR) {
+      Utils.tryMatch(
+        gb_2a, gb_2q, aliasMap).flatMap {
+        case g: GroupBy =>
+          Some(g.copy(child = g.child.withNewChildren(
+            g.child.children.map {
+              case modular.Select(_, _, _, _, _, _, _, _, _, _) => gb_2a;
+              case other => other
+            }), dataMapTableRelation =
+            if (isInheritTableRelation) gb_2a.dataMapTableRelation
```

Comment:
It is sort of hack, we should not copy the `dataMapTableRelation` like this. Actual matching should happen and the whole plan should be replaced not just `dataMapTableRelation`.